### PR TITLE
Add Apache Spark receiver

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

Add Apache Spark receiver to the Splunk distro, as well as integration tests. Some more work may be necessary to ensure docker image is pushed to quay repo.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### 🚀 New components 🚀
+
+- (Splunk) Add `apachespark` receiver ([#]())
```
Comment (crobert-1) [inline CHANGELOG.md:7]:
```suggestion
- (Splunk) Add `apachespark` receiver ([#5318](https://github.com/signalfx/splunk-otel-collector/pull/5318))
```

File: docs/components.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ The distribution offers support for the following components.
 | [awscontainerinsights](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/awscontainerinsightreceiver)      | [beta]           |
 | [awsecscontainermetrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/awsecscontainermetricsreceiver) | [beta]           |
```
Comment (pjanotti) [inline docs/components.md:18]: nit: not from your change, but, should be in alphabetical order by component name.

**Review by pjanotti — APPROVED** [review]

File: docs/components.md (hunk 2 above)
Comment (pjanotti) [reply docs/components.md:18]: Same for the actual go source code.

Comment (crobert-1) [reply docs/components.md:18]: I saw a few of these are out of order and was planning another PR to fix. I'll do that now in another PR, and rebase once the other once gets merged. 👍

Comment (crobert-1) [reply docs/components.md:18]: https://github.com/signalfx/splunk-otel-collector/pull/5321

**Review by jvoravong — APPROVED** [review]

File: internal/components/components.go [hunk 3]
```diff
@@ -164,6 +165,7 @@ func Get() (otelcol.Factories, error) {
 	receivers, err := receiver.MakeFactoryMap(
 		activedirectorydsreceiver.NewFactory(),
 		apachereceiver.NewFactory(),
+    apachesparkreceiver.NewFactory(),
```
Comment (crobert-1) [inline internal/components/components.go:168]:
```suggestion
		apachesparkreceiver.NewFactory(),
```

File: docker/apachespark/long_running.py
Comment (crobert-1) [inline docker/apachespark/long_running.py:1]: Note: This file was a direct copy from [here.](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f9704216a980405a35965d7d39400cb06a742c56/receiver/apachesparkreceiver/testdata/integration/sparkprograms/long_running.py)

File: docker/apachespark/Dockerfile
Comment (crobert-1) [inline docker/apachespark/Dockerfile:1]: Note: This file was a direct copy from [here.](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f9704216a980405a35965d7d39400cb06a742c56/receiver/apachesparkreceiver/testdata/integration/Dockerfile.apache-spark)

**Pull request merged**